google-cloud-pub-sub-grpc/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/grpc/PubSubSettings.scala

```diff
@@ -63,14 +74,21 @@ object PubSubSettings {
   /**
    * Create settings from config instance.
    */
-  def apply(config: Config): PubSubSettings =
+  def apply(config: Config)(implicit system: ClassicActorSystemProvider): PubSubSettings =
```

Comment:
@seglo This is a breaking change to address your comment https://github.com/akka/alpakka/pull/2613#discussion_r599848241. We need the system to create credentials.

The non-breaking option is to keep the dependency on Google's auth lib.

Reply:
See https://github.com/akka/alpakka/pull/2613#discussion_r600505557

In this case, maybe we can create a different factory method to avoid ambiguity with `apply`, and deprecate `apply` (we can replace it later).

Reply:
Reverted. The `system` would only have been necessary during the migration period: post-migration, credentials will no longer be handled by `PubSubSettings` anyway.